annotationsx/middleware.py

```diff
@@ -134,4 +134,23 @@ def process_response(self, request, response):
                                 path=settings.SESSION_COOKIE_PATH,
                                 secure=settings.SESSION_COOKIE_SECURE or None,
                                 httponly=settings.SESSION_COOKIE_HTTPONLY or None)
-        return response
\ No newline at end of file
+        return response
+
+
+class CookielessSessionMiddleware(object):
+    def __init__(self):
+        engine = import_module(settings.SESSION_ENGINE)
+        self.SessionStore = engine.SessionStore
+        print "STarting engine"
+
+    def process_request(self, request):
+        session_key = request.COOKIES.get(settings.SESSION_COOKIE_NAME, request.GET.get('utm_source'))
```

Comment:
I'll add the following to that line as a comment: "# note that 'utm_source' is an objuscated term that in actuality contains the session id injected into the URL when the user does not allow cookies"